examples/tv-app/linux/include/wake-on-lan/WakeOnLanManager.cpp

```diff
@@ -19,24 +19,41 @@
 #include "WakeOnLanManager.h"
 #include <fstream>
 #include <iostream>
+#include <platform/ConfigurationManager.h>
 #include <string>
 
 using namespace chip;
 using namespace chip::app::Clusters::WakeOnLan;
 
 std::string getMacAddress()
 {
-    std::ifstream input("/sys/class/net/eth0/address");
-    std::string line;
-    std::getline(input, line);
-    return line;
+    uint8_t macBuffer[chip::DeviceLayer::ConfigurationManager::kPrimaryMACAddressLength];
+    MutableByteSpan mac(macBuffer);
+    if (chip::DeviceLayer::ConfigurationMgr().GetPrimaryMACAddress(mac) != CHIP_NO_ERROR)
+    {
+        ChipLogProgress(Zcl, "WakeOnLanManager::getMacAddress no primary MAC configured by DeviceLayer");
+        return "0000000000";
+    }
+
+    char macStr[chip::DeviceLayer::ConfigurationManager::kPrimaryMACAddressLength * 2 + 1] = { 0 }; // added null char
+
+    uint8_t j = 0;
+    for (uint8_t i = 0; i < sizeof(macBuffer); i++)
+    {
+        j         = i * 2;
+        macStr[j] = (((macBuffer[i] & 0xF0) >> 4) & 0xF);
+        macStr[j] += (macStr[j] <= 9) ? '0' : ('A' - 10);
+        j++;
+        macStr[j] = (macBuffer[i] & 0x0F);
+        macStr[j] += (macStr[j] <= 9) ? '0' : ('A' - 10);
```

Comment:
We have existing BytesToHex helpers, no?

Reply:
Updated to use BytesToHex helper function - thanks!